spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UnicastSendingMessageHandler.java

```diff
@@ -167,8 +209,9 @@ protected final void setReliabilityAttributes(boolean lengthCheck,
 			this.ackTimeout = ackTimeout;
 		}
 		this.acknowledge = acknowledge;
-		if (this.acknowledge) {
+		if (!this.acknowledge) {
 			Assert.hasLength(ackHost);
+			Assert.state(StringUtils.isEmpty(socketExpression), "acknowledge cannot be used together with socketExpression");
```

Comment:
About that `if` - I swear I have no idea how this happend! Fortunately you are very perceptive :relieved: